packages/backend-core/src/features/index.ts

```diff
@@ -53,38 +120,84 @@ function isFlagName(name: string): name is keyof Flags {
  * they will be accessed through this function as well.
  */
 export async function fetch(): Promise<Flags> {
-  const currentTenantId = context.getTenantId()
-  const flags = defaultFlags()
-
-  const split = (env.TENANT_FEATURE_FLAGS || "")
-    .split(",")
-    .map(x => x.split(":"))
-  for (const [tenantId, ...features] of split) {
-    if (!tenantId || (tenantId !== "*" && tenantId !== currentTenantId)) {
-      continue
-    }
+  return await tracer.trace("features.fetch", async span => {
+    const tags: Record<string, any> = {}
+
+    const flags = defaultFlags()
 
-    for (let feature of features) {
-      let value = true
-      if (feature.startsWith("!")) {
-        feature = feature.slice(1)
-        value = false
+    const currentTenantId = context.getTenantId()
+    const specificallySetFalse = new Set<string>()
+
+    const split = (env.TENANT_FEATURE_FLAGS || "")
+      .split(",")
+      .map(x => x.split(":"))
+    for (const [tenantId, ...features] of split) {
+      if (!tenantId || (tenantId !== "*" && tenantId !== currentTenantId)) {
+        continue
       }
 
-      if (!isFlagName(feature)) {
-        throw new Error(`Feature: ${feature} is not an allowed option`)
+      for (let feature of features) {
+        let value = true
+        if (feature.startsWith("!")) {
+          feature = feature.slice(1)
+          value = false
+          specificallySetFalse.add(feature)
+        }
+
+        if (!isFlagName(feature)) {
+          throw new Error(`Feature: ${feature} is not an allowed option`)
+        }
+
+        if (typeof flags[feature] !== "boolean") {
+          throw new Error(`Feature: ${feature} is not a boolean`)
+        }
+
+        // @ts-ignore
```

Comment:
Why this ignore?

Reply:
![CleanShot 2024-08-09 at 10 34 40@2x](https://github.com/user-attachments/assets/d316905d-7ed5-4e84-b639-b7bfd1b4dde7)

If you can help me figure out why this happens I would be very grateful.

Reply:
Mm.. ts generic typing bullshit. There are ways to solve this, but not much more elegant.
One thing it can be easily done is `flags = { ...flags, [feature]: value }`

Reply:
I'm reticent to add a copy just to please the type system. I feel confident enough in the tests to keep the ignore in.